src/snapred/backend/recipe/algorithm/DiffractionSpectrumWeightCalculator.py

```diff
@@ -82,6 +82,24 @@ def PyExec(self):
             self.groupIDs.append(groupPeakList.groupID)
             self.predictedPeaks[groupPeakList.groupID] = groupPeakList.peaks
 
+        if all(len(peaks) == 0 for peaks in self.predictedPeaks.values()):
+            self.mantidSnapper.CloneWorkspace(
+                "Cloning a weighting workspce...",
+                InputWorkspace=self.inputWorkspaceName,
+                OutputWorkspace=self.weightWorkspaceName,
+            )
+            self.mantidSnapper.executeQueue()
+            weight_ws = self.mantidSnapper.mtd[self.weightWorkspaceName]
+            numSpec = weight_ws.getNumberHistograms()
+            for index in range(numSpec):
+                zeroDataX = np.zeros_like(weight_ws.readX(index))
+                weight_ws.dataX(index)[:] = zeroDataX
+                zeroDataY = np.zeros_like(weight_ws.readY(index))
+                weight_ws.dataY(index)[:] = zeroDataY
+
+            self.setPropertyValue("WeightWorkspace", self.weightWorkspaceName)
+            return
+
```

Comment:
Is this branch necessary?  It's preferable not to have branches with separate returns